### PR TITLE
Upgrade Surefire from 2.22.2 to 3.0.0-M7

### PR DESCRIPTION
**Pull request by basil**
This PR upgrades Surefire from 2.22.2 to 3.0.0-M7, which is the latest version available when you visit https://maven.apache.org/surefire/maven-surefire-plugin/ and the same version used in our plugin POM and all other core components. See [the release notes](https://maven.apache.org/surefire/maven-surefire-plugin/history.html). Some details about this upgrade:

- 3.0.0-M1 is susceptible to [SUREFIRE-1593](https://issues.apache.org/jira/browse/SUREFIRE-1593) and is therefore unsuitable.
- Starting with 3.0.0-M4, the baseline JVM memory requirement for Surefire for the main Maven process (_not_ the forked Surefire JVM used for running the tests) appears to have increased by about 300 MB, a requirement which continues up until 3.0.0-M7.
- 3.0.0-M6 is susceptible to [SUREFIRE-2076](https://issues.apache.org/jira/browse/SUREFIRE-2076) and is therefore unsuitable.

The current PR upgrades us to 3.0.0-M7 by deleting the custom downgrade, therefore getting version 3.0.0-M7 from our parent POM like all other core components. I have tested the 3.0.0-M7 upgrade in a variety of memory configurations:

- With the current 800 MB heap, which consistently fails with out of memory errors.
- With a 1 GB heap, which also consistently fails with out of memory errors.
- With 1100, 1256, 1396, and 1536 MB heaps, with which I have not seen a single out of memory error and all of which I have tested at least twice so far.

This PR selects a 1100 MB heap (a 38% increase) for the Maven process, which I have tested twice without seeing a single out of memory error (the PR builds from this PR will serve as additional testing).

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6652"></a>

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
Fine with me

**Review by NotMyFault — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
